models/Document.php

```diff
@@ -1110,6 +1112,8 @@ public function setParentId($parentId)
     {
         $this->parentId = (int) $parentId;
         $this->parent = null;
+        $this->siblings = [];
```

Comment:
Can you can explain this? The parent doesn't change until save is called. Otherwise you would have to do the same thing in setPublished